cmd/git-sync/main.go

```diff
@@ -184,11 +207,26 @@ func main() {
 	// From here on, output goes through logging.
 	log.V(0).Infof("starting up: %q", os.Args)
 
+	// Startup webhooks goroutine
+	webhookTriggerChan := make(chan struct{}, 1)
+	if *flWebhookURL != "" {
+		webhook := Webhook{
+			URL:             *flWebhookURL,
+			Method:          *flWebhookMethod,
+			Success:         flWebhookStatusSuccess,
+			Timeout:         *flWebhookTimeout,
+			BackoffDuration: *flWebhookBackoffDuration,
+		}
+		go webhook.run(webhookTriggerChan)
+	}
+
 	initialSync := true
 	failCount := 0
 	for {
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(*flSyncTimeout))
-		if err := syncRepo(ctx, *flRepo, *flBranch, *flRev, *flDepth, *flRoot, *flDest); err != nil {
+		if changed, err := syncRepo(ctx, *flRepo, *flBranch, *flRev, *flDepth, *flRoot, *flDest); err != nil {
+			syncDuration.WithLabelValues("error").Observe(time.Now().Sub(start).Seconds())
```

Comment:
This depends on metrics, which are not in yet?

Reply:
botched merge, fixed